### ISSUE TITLE
How to test with our own datas?

### ISSUE THREAD

**Issue by xinzi2018** [tinyalpha/BPR#10]
If I want to use my own coarse-mask datasets for testing, do I need to use "https://github.com/open-mmlab/mmdetection/tree/master/configs/cityscapes" to generate maskrcnn_val/maskrcnn_test first?

**Comment by xinzi2018** [comment]
![image](https://user-images.githubusercontent.com/36668024/123365799-991dfa00-d5a9-11eb-8a16-5b112fdfa552.png)

**Comment by chenhang98** [comment]
Yes. Note that our method is **post-processing**, so you first need to have segmentation results of an arbitrary instance segmentation model.
You can use Mask R-CNN, or any other model.

**Comment by xinzi2018** [comment]
The picture I showed above is the parsing I got, but I don’t know how to use your code to run it smoothly, because you have entered many other file paths in your tools/inference.sh test file, such as `GT_JSON=datasets/cityscapes/annotations/instancesonly_filtered_gtFine_val.json \` 


/////////////////////////////////////////////////////////////////////////////////////
`IOU_THRESH=0.55 \
IMG_DIR=/raid/home/dailingna/datasets/cityscapes/leftImg8bit/val/ \
GT_JSON=/raid/home/dailingna/datasets/cityscapes/annotations/instancesonly_filtered_gtFine_val.json \
BPR_ROOT=. \
GPUS=4 \
sh tools/inference.sh configs/bpr/hrnet48_256.py checkpoints/hrnet48_256-cbf4922c.pth maskrcnn_val maskrcnn_val_refined3`

 I ran this  command successfully.

**Comment by chenhang98** [comment]
It is recommended that you first process the dataset into the cityscapes format.
The `instancesonly_filtered_gtFine_val.json` in cityscapes is used to provide image and category information.

**Comment by chenhang98** [comment]
By the way, our pre-trained model may not be suitable for processing other datasets.

**Comment by lixuekai2001** [comment]
> Yes. Note that our method is **post-processing**, so you first need to have segmentation results of an arbitrary instance segmentation model.
> You can use Mask R-CNN, or any other model.

I see your model is pytorch based. But my instance segmentation is generated from Matterport Mask RCNN which is tensorflow/keras version. Do they work together?

**Comment by chenhang98** [comment]
Yes, as long as the result format is consistent. See https://github.com/tinyalpha/BPR#prepare-patches-dataset-optional for details.

**Comment by lixuekai2001** [comment]
> By the way, our pre-trained model may not be suitable for processing other datasets.
How can we train our own datasets?

**Comment by chenhang98** [comment]
@lixuekai2001 You can convert the dataset to the cityscape format, and then follow https://github.com/tinyalpha/BPR#training

**Comment by keynecky** [comment]
> @lixuekai2001 You can convert the dataset to the cityscape format, and then follow https://github.com/tinyalpha/BPR#training

If I want to use inference_coco.sh to inference on my own coarse masks, do I need to replace the IMG_DIR and GT_JSON with my own datasets? Or keep these args unchanged?

**Comment by chenhang98** [comment]
@keynecky Yes, you need to replace them with your own ones.

**Comment by Ocean-ZHY** [comment]
How did you convert to maskval, I don’t know how to convert, and the naming rules for conversion, can you tell me? thanks
